### PR TITLE
Developer Guide update for Storage component

### PR DESCRIPTION
**Pull request by antonTan96**
# DG Updates
* Updated Storage Component in Design section
* Features - some modifications to help feature documentation. Added sequence diagram and documentation for storage vaildation

# Code Quality updates
* Abstracted checking task-assigned `Employee`s in a `Project` in its own function to reduce arrow-headed code.

### CODE REVIEW

File: docs/diagrams/AddPSequenceDiagram.puml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+participant ":LogicManager" as LogicManager LOGIC_COLOR
+participant ":TaskHubParser" as TaskHubParser LOGIC_COLOR
+participant "c:AddProjectCommand" as AddProjectCommand LOGIC_COLOR
+end box
+
+box Model MODEL_COLOR_T1
+participant "toAdd:Project" as Project MODEL_COLOR
+participant ":Model" as Model MODEL_COLOR
+end box
+[-> LogicManager : execute("addP pr/Project1 em/1")
```
Comment (aslam341) [inline docs/diagrams/AddPSequenceDiagram.puml:15]: Supposed to be `addP n/Project1 em/1`.

File: docs/diagrams/AddPSequenceDiagram.puml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+activate LogicManager
+
+LogicManager -> TaskHubParser : parseCommand("addP pr/Project1 em/1")
```
Comment (aslam341) [inline docs/diagrams/AddPSequenceDiagram.puml:18]: Supposed to be `addP n/Project1 em/1`.

**Review by aslam341 — CHANGES_REQUESTED** [review]
There seems to be wrong command format usage in the diagrams.

File: src/main/java/seedu/address/storage/JsonAdaptedProject.java [hunk 3]
```diff
@@ -93,11 +93,7 @@ public Project toModelType() throws IllegalValueException {
         try {
             for (JsonAdaptedTask task : tasks) {
                 Task modeledTask = task.toModelType();
-                for (Employee assignedEmployee : modeledTask.getEmployee()) {
-                    if (!employeeList.strictlyContains(assignedEmployee)) {
-                        throw new IllegalValueException(MESSAGE_NOT_ASSIGNED_EMPLOYEE);
-                    }
-                }
+                checkEmployees(modeledTask.getEmployee(), employeeList);
```
Comment (rayyan35p) [inline src/main/java/seedu/address/storage/JsonAdaptedProject.java:96]: Good job in extracting the method to apply SLAP! 👍

**Review by rayyan35p — CHANGES_REQUESTED** [review]
As mentioned by Aslam, the commands are using the wrong prefix. Also, maybe you can add multiple indexes in the command to show what the loop in the sequence diagram will be going through? Just a suggestion.

**Review by rayyan35p — APPROVED** [review]
LGTM!

**Review by aslam341 — APPROVED** [review]

**Pull request merged**